### PR TITLE
Docs: Add documentation for the new encryption providers

### PR DESCRIPTION
**Pull request by undef1nd**
**What this PR does / why we need it**:
Adds documentation for the new key encryption key providers:
- Google Cloud KMS
- Hashicorp Key Vault

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/enterprise/kms-integration/using-google-cloud-kms-to-encrypt-database-secrets.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# Using Google Cloud KMS to encrypt database secrets
+
+You can use an encryption key from Google Cloud Key Management Service to encrypt secrets in the Grafana database.
+
+**Prerequisites:**
+
+- A Google Cloud account with permission to list and create KMS keys and service accounts to access those keys
+- Access to the Grafana [configuration]({{< relref "../../administration/configuration/#config-file-locations" >}}) file
+
+1. [Create a key ring](https://cloud.google.com/kms/docs/creating-keys#kms-create-key-ring-console).
```
Comment (mjseaman) [inline docs/sources/enterprise/kms-integration/using-google-cloud-kms-to-encrypt-database-secrets.md:17]:
```suggestion
1. [Create a key ring](https://cloud.google.com/kms/docs/creating-keys#kms-create-key-ring-console) in Google Cloud KMS.
```

File: docs/sources/enterprise/kms-integration/using-google-cloud-kms-to-encrypt-database-secrets.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+2. [Create a symmetric encryption key](https://cloud.google.com/kms/docs/creating-keys#create_a_key) in the key ring on the specified location.
```
Comment (mjseaman) [inline docs/sources/enterprise/kms-integration/using-google-cloud-kms-to-encrypt-database-secrets.md:19]: What does "specified location" mean?

File: docs/sources/enterprise/kms-integration/using-google-cloud-kms-to-encrypt-database-secrets.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+3. [Create a service account](https://cloud.google.com/iam/docs/creating-managing-service-accounts#creating) and assign it a role: it can be a predefined role or custom role with permissions to encrypt and decrypt secrets with Key Management Service.
+
+4. [Create and download a JSON file containing service account credentials](https://cloud.google.com/iam/docs/creating-managing-service-account-keys#creating).
```
Comment (mjseaman) [inline docs/sources/enterprise/kms-integration/using-google-cloud-kms-to-encrypt-database-secrets.md:23]: Where should I put this file?

File: docs/sources/enterprise/kms-integration/using-google-cloud-kms-to-encrypt-database-secrets.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+5. From within Grafana, turn on [envelope encryption]({{< relref "../../administration/envelope-encryption.md" >}}).
+
+6. Add your Google Cloud KMS details to the Grafana configuration file; depending on your operating system, is usually named `grafana.ini`:
+   <br><br>a. Add a new section to the configuration file, with a name in the format of `[security.encryption.azurekv.<KEY-NAME>]`, where `<KEY-NAME>` is any name that uniquely identifies this key among other provider keys.
+   <br><br>b. Fill in the section with the following values:
+   <br>
+
+   - `key_id`: encryption key ID, refer to [Getting the ID for a Key](https://cloud.google.com/kms/docs/getting-resource-ids#getting_the_id_for_a_key_and_version).
+   - `credentials_file`: path to the JSON file containing a service account credentials.
```
Comment (mjseaman) [inline docs/sources/enterprise/kms-integration/using-google-cloud-kms-to-encrypt-database-secrets.md:33]: Is this path relative to the config file itself?

File: docs/sources/enterprise/kms-integration/using-hashicorp-key-vault-to-encrypt-database-secrets.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+# Using Google Cloud KMS to encrypt database secrets
+
+You can use an encryption key from Google Cloud Key Management Service to encrypt secrets in the Grafana database.
+
+**Prerequisites:**
+
+- Permissions to manage Hashicorp Vault to enable secrets engines and issue tokens.
+- Access to the Grafana [configuration]({{< relref "../../administration/configuration/#config-file-locations" >}}) file
+
+1. [Enable the transit secrets engine](https://www.vaultproject.io/docs/secrets/transit#setup).
```
Comment (mjseaman) [inline docs/sources/enterprise/kms-integration/using-hashicorp-key-vault-to-encrypt-database-secrets.md:17]:
```suggestion
1. [Enable the transit secrets engine](https://www.vaultproject.io/docs/secrets/transit#setup) in Hashicorp Vault.
```

**Review by mjseaman — CHANGES_REQUESTED** [review]
Just a few comments and otherwise this looks good!

File: docs/sources/enterprise/kms-integration/using-google-cloud-kms-to-encrypt-database-secrets.md (hunk 2 above)
Comment (undef1nd) [reply docs/sources/enterprise/kms-integration/using-google-cloud-kms-to-encrypt-database-secrets.md:19]: Need to remove these bits. Thanks!

File: docs/sources/enterprise/kms-integration/using-google-cloud-kms-to-encrypt-database-secrets.md (hunk 3 above)
Comment (undef1nd) [reply docs/sources/enterprise/kms-integration/using-google-cloud-kms-to-encrypt-database-secrets.md]: Clarified with example.

File: docs/sources/enterprise/kms-integration/using-google-cloud-kms-to-encrypt-database-secrets.md (hunk 4 above)
Comment (undef1nd) [reply docs/sources/enterprise/kms-integration/using-google-cloud-kms-to-encrypt-database-secrets.md]: Clarified.

**Comment by osg-grafana** [comment]
Thank yous for your work on this. It means a lot to me that this has not fallen through the cracks. :)

**Review by mjseaman — APPROVED** [review]

**Pull request merged**